### DISCUSSION TITLE
`input.showPicker()`

### DISCUSSION THREAD

**Discussion by LeaVerou** [Devographics/surveys#153] [State of HTML Features]
Programmatically open the picker of form controls that have one (color pickers, date inputs etc).

### Code sample

```html
<input id="dateInput" type="date">
<button onclick="dateInput.showPicker()">Select date</button>
```

### MDN URL:

https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/showPicker

### Specification URL:

https://html.spec.whatwg.org/multipage/input.html#dom-input-showpicker

### Any other links?

- [TAG review](https://github.com/w3ctag/design-reviews/issues/688)

### How stable is this feature?

Cutting edge (Shipped, but not in all browsers)

### Type of feature

JS API

### What focus area does this feature fall under?

Forms

**Comment by domenic** [comment]
> Cutting edge (Shipped, but not in all browsers)

[According to MDN](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/showPicker) it's in all browsers, although with some rough edges in Firefox and Safari.

**Reply by LeaVerou** [reply]
It seems to only support a subset of input types in Firefox, and even fewer in Safari (it's unclear what *can* actually do in Safari)
